Src/Logic/Registration/Registration.ts

```diff
@@ -0,0 +1,60 @@
+import generateFirstStepForm from '../../UIComponents/FormComponent/FormComponent';
+import { User } from '../../../Models/User.model';
+import { FirstStepFormValues } from '../../../Src/UIComponents/FormComponent/FormComponent';
+import { createElement } from '../../../Src/UIComponents/utils/utils';
+import {
+  SecondStepValues,
+  generateSecondStep,
+} from '../../../Src/UIComponents/SecondStepComponent/SecondStepComponent';
+import { Weight } from '../../../Models/Weight.model';
+import {
+  readFromLocalStorage,
+  saveInLocalStorage,
+} from '../../../Src/Logic/LocalStorage/LocalStorage';
+
+export default function generateRegistrationForm(
+  onRegistrationSuccess: () => void,
+): HTMLElement {
+  const registrationDiv = createElement('div', 'div-registration');
+
+  let firstStepValues: FirstStepFormValues;
+  const firstStepForm = generateFirstStepForm(onNextStepClick);
+  firstStepForm.className = 'first-step-form';
+  registrationDiv.appendChild(firstStepForm);
+  const secondStepForm = generateSecondStep(onBackClick, onSignUpClick);
+
+  function onNextStepClick(firstStepFormValues: FirstStepFormValues): void {
+    firstStepValues = firstStepFormValues;
+    registrationDiv.replaceChild(secondStepForm, firstStepForm);
+  }
+
+  function onBackClick() {
+    registrationDiv.replaceChild(firstStepForm, secondStepForm);
+  }
+
+  function onSignUpClick(secondStepValues: SecondStepValues) {
+    const weight: Weight = {
+      date: new Date(),
+      weight: firstStepValues.currentWeight,
+    };
+    const user: User = {
+      name: firstStepValues.name,
+      gender: firstStepValues.gender,
+      dateOfBirth: firstStepValues.dateOfBirth,
+      height: firstStepValues.height,
+      goalWeight: firstStepValues.goalWeight,
+      activityLevel: secondStepValues.activityLevel,
+      weights: [weight],
+    };
+
+    const userFromLocalStorage = readFromLocalStorage(user.name);
+    if (userFromLocalStorage) {
+      alert('This nickname is taken. Please select another.');
```

Comment:
maybe instead of alert, some message in the component could be displayed? (it's more of a UX suggestion than a code review one)

Reply:
I have removed alert and changed to modal window. Konrad created radio buttons and told ,that he will fix them, when this pull request will be merged.